pallet-logion-loc/src/lib.rs

```diff
@@ -717,21 +729,29 @@ pub mod pallet {
                 Err(Error::<T>::NotFound)?
             } else {
                 let loc = <LocMap<T>>::get(&loc_id).unwrap();
-                if loc.owner != who {
+                let submitted_by_owner: bool = loc.owner == who;
+                if !submitted_by_owner && !Self::can_submit(&loc_id, &loc, &SupportedAccountId::Polkadot(who.clone())) {
                     Err(Error::<T>::Unauthorized)?
+                } else if !submitted_by_owner && item.submitter != SupportedAccountId::Polkadot(who) {
```

Comment:
To me, this condition seems a bit redundant with previous one. Is it important to make a difference between unauthorized and invalid?

Reply:
Well both tests need to be done, the first one check the authorization while the second one checks the data consistency. But I can merge them together, indeed.